sdk/data/azcosmos/cosmos_client_retry_policy.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type clientRetryPolicy struct {
+	gem                    *globalEndpointManager
+	retryCount             int
+	sessionRetryCount      int
+	preferredLocationIndex int
+}
+
+const maxRetryCount = 10 // this number needs to be higher, keeping low for testing purposes right now
+const defaultBackoff = 1
+
+func (p *clientRetryPolicy) Do(req *policy.Request) (*http.Response, error) {
+	response, err := req.Next() // err can happen in weird scenarios (connectivity, etc) - need to test
+	o := pipelineRequestOptions{}
+	req.OperationValue(&o)
```

Comment:
What is this call for?

Reply:
I got this from the header policy, but it populates the previously empty pipelineRequestOptions with those that are present in the request

Reply:
https://github.com/Azure/azure-sdk-for-go/blob/10dd3c522da479beb9dc1330c6863b7cebcc6e18/sdk/data/azcosmos/cosmos_headers_policy.go#L24-L27

The header policy is using this to **extract** the values from the Request. You need to test if this returns `true` before using any values

Reply:
ok makes sense - what should we do in case the returned boolean is false? if we default `isWriteOperation` to false then the current logic could stand no? or should we instead make the logic ourselves and determine if it is a write request based on the request method + parsing of the url?